File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        string ans = "";
        int counter = 0;
        for(int i=0;i<s.length();i++){
            if(s[i] == '('){
                if(counter>0){
                    ans += s[i];
                }
                counter++;
            }
            if(s[i] == ')'){
                counter--;
                if(counter>0){
                    ans += s[i];
                }
            }
        }
        return ans;
    }
};

File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        Stack<Character> st = new Stack<>();
        StringBuilder ans = new StringBuilder();
        for (char ch : s.toCharArray()) {
            if (ch == '(') {
                if (st.isEmpty()) {
                    st.push(ch);
                }
                else if (!st.isEmpty()) {
                    st.push(ch);
                    ans.append(ch);
                }
            }

            else{
                st.pop();
                if(!st.isEmpty()){
                    ans.append(')');
                }
            }
        }
        return ans.toString();
    }
}